src/types/orchestrationTypes.ts

```diff
@@ -65,3 +85,112 @@ export interface CallHttpOptions {
      */
     asynchronousPatternEnabled?: boolean;
 }
+
+/**
+ * Represents the status of a durable orchestration instance.
+ *
+ * Can be fetched using `DurableClient.getStatus()`
+ */
+export declare class DurableOrchestrationStatus {
+    /**
+     * The orchestrator function name.
+     */
+    readonly name: string;
+
+    /**
+     * The unique ID of the instance.
+     *
+     * The instance ID is generated and fixed when the orchestrator
+     * function is scheduled. It can either auto-generated, in which case
+     * it is formatted as a GUID, or it can be user-specified with any
+     * format.
+     */
+    readonly instanceId: string;
+
+    /**
+     * The time at which the orchestration instance was created.
+     *
+     * If the orchestration instance is in the [[Pending]] status, this
+     * time represents the time at which the orchestration instance was
+     * scheduled.
+     */
+    readonly createdTime: Date;
+
+    /**
+     * The time at which the orchestration instance last updated its
+     * execution history.
+     */
+    readonly lastUpdatedTime: Date;
+
+    /**
+     * The input of the orchestration instance.
+     */
+    readonly input: unknown;
+
+    /**
+     * The output of the orchestration instance.
+     */
+    readonly output: unknown;
+
+    /**
+     * The runtime status of the orchestration instance.
+     */
+    readonly runtimeStatus: OrchestrationRuntimeStatus;
+
+    /**
+     * The custom status payload (if any) that was set by
+     * `DurableOrchestrationClient.setCustomStatus()`.
+     */
+    readonly customStatus?: unknown;
+
+    /**
+     * The execution history of the orchestration instance.
+     *
+     * The history log can be large and is therefore `undefined` by
+     * default. It is populated only when explicitly requested in the call
+     * to `DurableOrchestrationClient.getStatus()`.
+     */
+    readonly history?: Array<unknown>;
+}
+
+/**
+ * The status of an orchestration instance.
+ */
+export enum OrchestrationRuntimeStatus {
```

Comment:
enums are more like classes than interfaces. enums & classes stick around when compiled to javascript, whereas types & interfaces are thrown away. That means you still need the real code in the src folder, and this file can only do `export declare enum`. It works now because this file is a `.ts` file, but it should break once this file is converted to a `.d.ts` file

In addition, all imports of `OrchestrationRuntimeStatus` will need to reference the src file, not the types file.

Reply:
Done. Is there an equivalent of the `implements` here, or how is the `.d.ts` declaration mapped to the src file?

Reply:
I don't think there's an `implements`, no. Tbh I don't use enums very often as they can be kinda clunky